src/main/java/mil/dds/anet/beans/Report.java

```diff
@@ -786,6 +791,16 @@ public boolean isFutureEngagement() {
     return engagementDate != null && engagementDate.isAfter(Utils.endOfToday());
   }
 
+  @GraphQLQuery(name = "engagementStatus")
+  public List<EngagementStatus> loadEngagementStatus() {
+    LinkedList<EngagementStatus> statuses = new LinkedList<EngagementStatus>();
+    if (ReportState.CANCELLED.equals(getState())) {
```

Comment:
Move `state` from `ReportDao::additionalFields` to `ReportDao::minimalFields` (else a GraphQL search that requests `engagementStatus` but not `state` will give wrong results; note that `engagementDate` [used in `isFutureEngagement()`] is already part of the `minimalFields`).
Also, `ReportState.CANCELLED.equals(getState())` is better expressed as `state == ReportState.CANCELLED` (it's an enum).